src/util/support/threads.c

```diff
@@ -151,7 +150,6 @@ int krb5int_pthread_loaded (void)
         || &pthread_mutex_destroy == 0
         || &pthread_mutex_init == 0
         || &pthread_self == 0
-        || &pthread_equal == 0
```

Comment:
On some glibc versions, [`pthread_equal()` may be defined as an inline function](https://sourceware.org/git/?p=glibc.git;a=blobdiff;f=nptl/sysdeps/pthread/pthread.h;h=f60ecdee18c48559603baec1e5fceb0534f38a8b;hp=5f34302e0c23a2b40301a35d67d2421064519a24;hb=c26ca5e17ded3275872d9fb48776d155ca7285d5;hpb=1b8373f475105307ee3b64d423ffec995ddd6cde) in the pthread header file. It would result in this test to always be true, no matter if libpthread is loaded or not.